File: src/main/java/org/poglebiarka/app/security/SecurityConfiguration.java
package org.poglebiarka.app.security;

import com.vaadin.flow.spring.security.VaadinWebSecurity;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.web.util.matcher.AntPathRequestMatcher;

@EnableWebSecurity
@Configuration
public class SecurityConfiguration extends VaadinWebSecurity {

    @Override
    protected void configure(HttpSecurity http) throws Exception {

        http.authorizeHttpRequests(
                authorize -> authorize.requestMatchers(new AntPathRequestMatcher("/images/*.png")).permitAll());
        // Icons from the line-awesome addon
        http.authorizeHttpRequests(authorize -> authorize
                .requestMatchers(new AntPathRequestMatcher("/line-awesome/**/*.svg")).permitAll());

        setOAuth2LoginPage(http, "/login");
        http.logout( cfg -> cfg.logoutSuccessUrl("/login").permitAll());

        super.configure(http);
    }

}

File: src/main/kotlin/org/poglebiarka/admin/security/SecurityConfiguration.kt
package org.poglebiarka.admin.security;

import com.vaadin.flow.spring.security.VaadinWebSecurity;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.web.util.matcher.AntPathRequestMatcher;

@EnableWebSecurity
@Configuration
class SecurityConfiguration : VaadinWebSecurity() {

    override fun configure(http: HttpSecurity) {

        http.authorizeHttpRequests {
            it.requestMatchers(AntPathRequestMatcher("/images/*.png")).permitAll()
        }

        http.authorizeHttpRequests {
                it.requestMatchers(AntPathRequestMatcher("/line-awesome/**/*.svg")).permitAll()
        }

        setOAuth2LoginPage(http, "/login")
        http.logout {
            it.logoutSuccessUrl("/login").permitAll()
        }

        super.configure(http)
    }

}
